cs3/app/registry/v1beta1/resources.proto

```diff
@@ -42,23 +42,54 @@ message ProviderInfo {
   // The address where the app provider can be reached.
   // For example, tcp://localhost:1099.
   string address = 3;
+  // REQUIRED.
+  // The capability of the underlying app.
+  enum Capability {
+    CAPABILITY_INVALID = 0;
+    // The app is a simple viewer.
+    CAPABILITY_VIEWER = 1;
+    // The app is a full editor.
+    CAPABILITY_EDITOR = 2;
```

Comment:
How exactely would the UI make use of the capabilities? 
If it has `CAPABILITY_VIEWER` obviously shouldn't offer to edit a file... But if something is a `CAPABILITY_EDITOR` a user might also want to know if it's possible to open a file in reading mode.
Shouldn't the Capabilities reflect all possible OpenInAppRequest.ViewMode(s) https://cs3org.github.io/cs3apis/#cs3.gateway.v1beta1.OpenInAppRequest.ViewMode?

Reply:
This was discussed in a now merged PR. There are two motivations for this:
* In the context menu for files, one could show `Preview in <app>` vs both `Open in <app>` and `Preview`. Of course if the user has no write access, only `Preview` would be shown.
* Populate the "New" button menu with mime types for which an AppProvider is registered with `EDITOR` capability. As a (counter-)example, assuming that e.g. Collabora is able to open jpeg files in view mode, do not show it in the New list.

Now the first is not super urgent in the current state of integrations, whereas the second is quite critical. The other part of this PR - providing a list of "well-known" mime types with their description and corresponding file extensions - specifically targets that.

Finally, I did think about some "beauty of symmetry" to make `capabilities` reflect the `viewMode` values, but I don't see the added value - for the UI you just need to know if an app is capable of editing or not, a viewer with or without a download button (and with or without the ability to hide it should the permission be so restrictive) is still a viewer and would be fired from a `Preview` menu item.

Reply:
Another upfront consideration is the fact that we do have a number of "UI-only" apps (e.g. drawio, the pdfviewer, etc.) which are not captured by the AppProviders scheme. So anyway the UI must take those into account differently.

Reply:
> Another upfront consideration is the fact that we do have a number of "UI-only" apps (e.g. drawio, the pdfviewer, etc.) which are not captured by the AppProviders scheme. So anyway the UI must take those into account differently.

That's true but these apps are also ownCloud Web specific and can't be used in any other web frontend. The app driver returns generic application which can also be used in a different web frontend. Therefore it is ok to just list what the backend / reva provides.